include/ignition/rendering/Camera.hh

```diff
@@ -282,6 +282,12 @@ namespace ignition
       /// \return Texture Id of type GLuint.
       public: virtual unsigned int RenderTextureGLId() const = 0;
 
+      /// \brief Notifies us that the shadow node definition is about to be
+      /// updated. This means our compositor workspace must be destroyed
+      /// because the shadow node definition it's using will become a
+      /// dangling pointer otherwise
+      public: virtual void SetShadowsNodeDefDirty() = 0;
```

Comment:
in ignition we guarantee ABI compatibility for released versions. So adding a virtual function here unfortunately breaks ABI. Can you move this function to Ogre2Camera and Ogre2DepthCamera (I think those are the only 2 classes that need them) and make them non virtual? Then in `Ogre2Scene::PreRender`, we should be able to just call the derived class' `SetShadowsNodeDefDirty` directly.

Reply:
Ouch!!! (Swears internally)

Then we can implement this version on unstable branch, while stable uses downcasting and calling the function directly

Reply:
How do you normally handle this?

#ifdefs e.g.

```
#if IGN_VERSION > 4
public: virtual void SetShadowsNodeDefDirty() = 0;
#endif

#if IGN_VERSION > 4
camera->SetShadowsNodeDefDirty();
#else
 // downcast version
#endif
```

or just each branch its own piece of code?

Reply:
> or just each branch its own piece of code?


yeah we'll need to have different code. The process goes like:
1. add downcast version in ign-rendering5 + add a note in appropriate places, e.g. in Ogre2Camera and Ogre2DepthCamera's `SetShadowsNodeDefDirty` function to mention that this is done for ABI compatibility.
2. When this PR gets merged, someone (e.g. me) will do a merge forward from ign-rendering5 to the unstable branch, `main`
3. During the merge forward, I can take care of changing the implementation to use this version